lib/rules/no-container.ts

```diff
@@ -0,0 +1,103 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils';
+import { getDocsUrl } from '../utils';
+import {
+  isIdentifier,
+  isMemberExpression,
+  isObjectPattern,
+  isProperty,
+  isRenderVariableDeclarator,
+} from '../node-utils';
+
+export const RULE_NAME = 'no-container';
+
+export default ESLintUtils.RuleCreator(getDocsUrl)({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow the use of container methods',
+      category: 'Best Practices',
+      recommended: 'error',
+    },
+    messages: {
+      noContainer:
+        'Avoid using container to query for elements. Prefer using query methods from Testing Library, such as "getByRole()"',
+    },
+    fixable: null,
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          renderFunctions: {
+            type: 'array',
+          },
+        },
+      },
+    ],
+  },
+  defaultOptions: [
+    {
+      renderFunctions: [],
+    },
+  ],
+
+  create(context, [options]) {
+    const { renderFunctions } = options;
+    let containerName = '';
+    let renderWrapperName = '';
+    let hasPropertyContainer = false;
+
+    return {
+      VariableDeclarator(node) {
+        if (isRenderVariableDeclarator(node, renderFunctions)) {
+          if (isObjectPattern(node.id)) {
+            const containerIndex = node.id.properties.findIndex(
+              property =>
+                isProperty(property) &&
+                isIdentifier(property.key) &&
+                property.key.name === 'container'
+            );
+            const nodeValue =
+              containerIndex !== -1 && node.id.properties[containerIndex].value;
+            containerName = isIdentifier(nodeValue) && nodeValue.name;
+          } else {
+            renderWrapperName = isIdentifier(node.id) && node.id.name;
+          }
+        }
+      },
+
+      CallExpression(node: TSESTree.CallExpression) {
+        function showErrorForChainedContainerMethod(
+          innerNode: TSESTree.MemberExpression
+        ) {
+          if (isMemberExpression(innerNode)) {
+            if (isIdentifier(innerNode.object)) {
+              const isScreen = innerNode.object.name === 'screen';
```

Comment:
we should drop this line. `screen` does not expose the `container` property - see the types [here](https://github.com/testing-library/dom-testing-library/blob/master/types/screen.d.ts). Sorry if I caused confusion when mentioning it before! My bad

Reply:
Sure! No problem :)